pkg/download/downloader.go

```diff
@@ -58,15 +63,46 @@ func Download(d Downloader) (int, io.ReadCloser, error) {
 		return resp.StatusCode, resp.Body, nil
 	}
 
-	err = fmt.Errorf("unexpected status code: actual=%d expected=%d", resp.StatusCode, http.StatusOK)
+	errString := ""
+	requestId := resp.Header.Get(xMsServiceRequestIdHeaderName)
 	switch d.(type) {
 	case *blobWithMsiToken:
 		switch resp.StatusCode {
 		case http.StatusNotFound:
-			return resp.StatusCode, nil, errors.Wrapf(err, MsiDownload404ErrorString)
+			errString = MsiDownload404ErrorString
 		case http.StatusForbidden:
-			return resp.StatusCode, nil, errors.Wrapf(err, MsiDownload403ErrorString)
+			errString = MsiDownload403ErrorString
 		}
+		break
+	default:
+		uri := req.URL.Host //"string"
+		switch resp.StatusCode {
+		case http.StatusUnauthorized:
+			errString = fmt.Sprintf("CustomScript failed to download the file from %s because access was denied. Please fix the blob permissions and try again, the response code and message returned were: %q",
+				uri,
+				resp.Status)
+		case http.StatusNotFound:
+			errString = fmt.Sprintf("CustomScript failed to download the file from %s because it does not exist. Please create the blob and try again, the response code and message returned were: %q",
+				uri,
+				resp.Status)
+
+		case http.StatusBadRequest:
+			errString = fmt.Sprintf("CustomScript failed to download the file from %s because parts of the request were incorrectly formatted, missing, and/or invalid. The response code and message returned were: %q",
+				uri,
+				resp.Status)
+
+		case http.StatusInternalServerError:
+			errString = fmt.Sprintf("CustomScript failed to download the file from %s due to an issue with storage. The response code and message returned were: %q",
+				uri,
+				resp.Status)
+		default:
+			errString = fmt.Sprintf("CustomScript failed to download the file from %s because the server returned a response code and message of %q Please verify the machine has network connectivity.",
+				uri,
```

Comment:
does uri also contain path? a typical uri looks like https://hostname/path?query
AFAIK scripts can be inside blob containers and container name is usually in the path.
I believe CSE creates a directory with the same name as the blob container before downloading the blob, that is in case blob is inside a container.

Reply:
uri only shows the host or host:port part of the request URL. I'll rename the variable since it's a bit misleading, but the path and everything after should be cut off